lib/src/logging_service.dart

```diff
@@ -75,4 +104,40 @@ class LoggingService {
   }
 
   bool shouldLog(String channel) => _enabledChannels.contains(channel);
+
+  /// Registers a service extension method with the given name and a callback to
+  /// be called when the extension method is called.
+  void _registerServiceExtension(
+      {@required String name, @required _ServiceExtensionCallback callback}) {
+    assert(name != null);
+    assert(callback != null);
+    final String methodName = 'ext.flutter.$name';
```

Comment:
nit: make this
ext.flutter.logs.$name

and you can simplify this code slightly